core/vm/contracts_stateful.go

```diff
@@ -0,0 +1,165 @@
+package vm
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/coreth/params"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/holiman/uint256"
+)
+
+// PrecompiledContractsApricot contains the default set of pre-compiled Ethereum
+// contracts used in the Istanbul release and the stateful precompiled contracts
+// added for the Avalanche Apricot release.
+// Apricot is incompatible with the YoloV1 Release since it does not include the
+// BLS12-381 Curve Operations added to the set of precompiled contracts
+var PrecompiledContractsApricot = map[common.Address]StatefulPrecompiledContract{
```

Comment:
What do the keys in this map represent? Is there some canonical ordering of these contracts defined somewhere?

Reply:
The keys in this map represent the addresses that the precompiled contracts can be called at. They are ordered in each of the ethereum hard forks as PrecompiledContractsIstanbul, PrecompiledContractsYoloV1, etc. The precompiled contracts for apricot match the precompiled contracts used in the latest mainnet fork, so that we support all of the precompiles in addition to the ones that we've added. Ours start at a higher address, so that they won't conflict with future additions.

Reply:
Can we record the above in a comment here?